src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -389,3 +490,359 @@ def names(self, names):
+                try:
+                    subprocess_cmd = kubectl_prior + ["get", "apd", apdName, "-n", "aks-periscope", "-o=jsonpath={.spec.networkconfig}"]
+                    network_config = subprocess.check_output(
+                        subprocess_cmd,
+                        universal_newlines=True)
+                    logger.debug('Dns status for node %s is %s', node_name, network_config)
+                    subprocess_cmd = kubectl_prior + ["get", "apd", apdName, "-n", "aks-periscope", "-o=jsonpath={.spec.networkoutbound}"]
+                    network_status = subprocess.check_output(
+                        subprocess_cmd,
+                        universal_newlines=True)
+                    logger.debug('Network status for node %s is %s', node_name, network_status)
+
+                    if not network_config or not network_status:
+                        print("The diagnostics information for node {} is not ready yet. "
+                              "Will try again in 10 seconds.".format(node_name))
+                        time.sleep(10)
+                        break
+
+                    network_config_array += json.loads('[' + network_config + ']')
+                    network_status_object = json.loads(network_status)
+                    network_status_array += format_diag_status(network_status_object)
+                    ready_nodes[node_name] = True
+                except subprocess.CalledProcessError as err:
+                    raise CLIInternalError(err.output)
+
+    print()
+    if network_config_array:
+        print("Below are the network configuration for each node: ")
+        print()
+        print(tabulate(network_config_array, headers="keys", tablefmt='simple'))
+        print()
+    else:
+        logger.warning("Could not get network config. "
+                       "Please run 'az connectedk8s troubleshoot' command again later to get the analysis results.")
+
+    if network_status_array:
+        print("Below are the network connectivity results for each node:")
+        print()
+        print(tabulate(network_status_array, headers="keys", tablefmt='simple'))
+    else:
+        logger.warning("Could not get networking status. "
+                       "Please run 'az connectedk8s troubleshoot' command again later to get the analysis results.")
+
+
+def format_diag_status(diag_status):
+    for diag in diag_status:
+        if diag["Status"]:
+            if "Error:" in diag["Status"]:
+                diag["Status"] = f'{colorama.Fore.RED}{diag["Status"]}{colorama.Style.RESET_ALL}'
+            else:
+                diag["Status"] = f'{colorama.Fore.GREEN}{diag["Status"]}{colorama.Style.RESET_ALL}'
+
+    return diag_status
+
+
+def collect_periscope_logs(resource_group_name, name, storage_account_name=None, sas_token=None, readonly_sas_token=None, kube_context=None, kube_config=None):
+    colorama.init()
+
+    if not which('kubectl'):
```

Comment:
Why does this have a dependency on kubectl?

Reply:
I think, this Sophie implemented this similar to what aks-periscope does. https://github.com/Azure/azure-cli-extensions/blob/a243ab0cdfc77570341063fa5aa3b7660e107e31/src/aks-preview/azext_aks_preview/custom.py#L1988